### PR TITLE
Conceivable sizes

### PR DESCRIPTION
**Pull request by dimwit105**
Closes DFHack/DFHack#4506

Adjusts the size line to compare units to something more reasonable than cubic centimeters. Will compare small things to cats, most animals and humanoids to dwarves, and particularly large animals to elephants. 

Also adds a line to show that particular unit's size modifiers.

### CODE REVIEW

File: gui/unit-info-viewer.lua
Comment (myk002) [inline gui/unit-info-viewer.lua:1]: please add a line about this improvement in `changelog.txt`

**Comment by myk002** [comment]
pre-commit.ci autofix

File: gui/unit-info-viewer.lua [hunk 1]
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
...
+    elseif size > base_comparison_size*0.25 then
+        return string.format('%.3f average dwarves', size/6000)
+    else
+        return string.format('%.3f average cats', size/500)
+    end
+
+end
+
+local function get_size_compared_to_median(unit)
+    local size_normal = unit.body.size_info.size_base
```
Comment (myk002) [inline gui/unit-info-viewer.lua:351]: unused variable?

File: gui/unit-info-viewer.lua [hunk 2]
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
     return unit.body.size_info.size_cur * 10
 end
 
+local function get_conceivable_comparison(unit, comparison_unit)
+    --the objective here is to get a (resaonably) small number to help concieve how large a thing is. "83 dwarves" doesn't really help convey the size of an elephant much better than 5m cc, so at certain breakpoints we will use different animals
```
Comment (myk002) [inline gui/unit-info-viewer.lua:337]:
```suggestion
    -- the objective here is to get a (reasonably) small number to help conceive
    -- how large a thing is. "83 dwarves" doesn't really help convey the size of an
    -- elephant much better than 5m cc, so at certain breakpoints we will use
    -- different animals
```

File: gui/unit-info-viewer.lua [hunk 3]
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
     return unit.body.size_info.size_cur * 10
```
Comment (myk002) [inline gui/unit-info-viewer.lua:333]: is this function now unused? if so, you can remove it

File: gui/unit-info-viewer.lua [hunk 4]
```diff
@@ -333,12 +333,42 @@ local function get_size_in_cc(unit)
...
+local function get_conceivable_comparison(unit, comparison_unit)
+    --the objective here is to get a (resaonably) small number to help concieve how large a thing is. "83 dwarves" doesn't really help convey the size of an elephant much better than 5m cc, so at certain breakpoints we will use different animals
+    local size = unit.body.size_info.size_cur
+    local base_comparison_size = 6000
+    if size > base_comparison_size*20 then
+        return string.format('%.3f average elephants', size/500000)
+    elseif size > base_comparison_size*0.25 then
+        return string.format('%.3f average dwarves', size/6000)
+    else
+        return string.format('%.3f average cats', size/500)
```
Comment (myk002) [inline gui/unit-info-viewer.lua:345]: I'm not sure we even need this much accuracy. How about `%.1f`?

File: gui/unit-info-viewer.lua [hunk 5]
```diff
@@ -451,6 +481,7 @@ function UnitInfo:refresh(unit, width)
     add_chunk(chunks, get_ghostly_chunk(unit), width)
     add_chunk(chunks, get_dead_chunk(unit), width)
     add_chunk(chunks, get_body_chunk(unit), width)
+    add_chunk(chunks, get_average_size(unit), width)
```
Comment (myk002) [inline gui/unit-info-viewer.lua:484]: maybe avoid adding this chunk if the race is dwarf, elephant, or cat. We already know the unit is average/small/large for its race; we don't need to know that the elephant weighs about as much as 1.0 elephants.

**Comment by dimwit105** [comment]
I've removed the extra function after checking it wasnt used anywhere else, and reformatted that long comment so it isnt a huge line anymore. 

I've adjusted the way weights are displayed, so elephants/cats arent reported as weighing 1 elephant or 1 cat, but instead display weight in dwarves. Dwarves themselves no longer give a precise weight, just the relative to average dwarfiness.

**Comment by myk002** [comment]
pre-commit.ci autofix

**Review by myk002 — COMMENTED** [review]
Looks great! Could you add a line to the "Misc Improvements" section for the "Future" version in `changelog.txt`?

**Comment by myk002** [comment]
Also, please feel free to add yourself to https://github.com/DFHack/dfhack/blob/develop/docs/about/Authors.rst (in a separate PR)

**Comment by dimwit105** [comment]
The changelog has been added. I have managed to bumble through my first GitHub experience, and my journey to GitHub Wizard has begun

**Review by myk002 — APPROVED** [review]

**Pull request merged**